### PR TITLE
Update image upload component

### PR DESCRIPTION
**Pull request by Huizz**
… and 

**Changes**
Description of changes...
- delete branch

**Changelog entry**

Image upload component
- Fix colour for image upload description
- Add drop files text
- Align "Add Photos" button the drop files text to right of component

**Additional information**

-   You may refer to figma for the intended design

### CODE REVIEW

**Comment by weili-govtech** [comment]
are the font weights and spacings part of this fix? I notice the title need to have a font-weight of 600 and 0.5rem margin to the description below.

**Comment by Huizz** [comment]
thanks for noticing. 

The font is already loaded as "Open Sans semibold" so it should already be weight 600. not sure why the developer console shows it as 400. 
actual font-weight 400 will look like this
image

**Comment by Huizz** [comment]
as for the spacing. there was this portion in the original styles for `<DragUpload />`
```
	> :not(:last-child) {
		margin-bottom: 1rem;
	}
```

I dont have a reference for when the component does not have the description. would you happen to know if this style is still valid and the 0.5rem only applies if there is a description?

**Comment by Huizz** [comment]
another curious point.

description is optional in `IImageUploadSchema` but required in `IImageInputProps`.
there is no fallback to a default value if description is undefined, before passing it to `<ImageInput />` as part of `IImageInputProps`

should description also be optional  in `IImageInputProps`?

**Comment by weili-govtech** [comment]
> The font is already loaded as "Open Sans semibold" so it should already be weight 600. not sure why the developer console shows it as 400.

whoops my bad, I missed that out, then this is fine.

---

> as for the spacing. there was this portion in the original styles for <DragUpload />

	> :not(:last-child) {
		margin-bottom: 1rem;
	}
> I dont have a reference for when the component does not have the description. would you happen to know if this style is still valid and the 0.5rem only applies if there is a description?

the quoted style for `<DragUpload />` is not meant for the description though. currently there have not been any use case for a component without description but I think we can remove that `0.5rem` if there are no descriptions.

---

> another curious point.

> description is optional in IImageUploadSchema but required in IImageInputProps.
there is no fallback to a default value if description is undefined in IImageUploadSchema, before passing it to <ImageInput /> as part of IImageInputProps

> should description also be optional in IImageInputProps?

yes that's a fair point, can update as optional.

---

note: I removed the figma link as this is a public repo, we can share the mocks through another medium instead.

**Comment by Huizz** [comment]
> the quoted style for <DragUpload /> is not meant for the description though. currently there have not been any use case for a component without description but I think we can remove that 0.5rem if there are no descriptions.

Hmm I styled it in a different way, with the `hasDescription` prop. Do let me know if that is okay

> yes that's a fair point, can update as optional.

Okay this has been updated too

File: src/components/fields/image-upload/image-input/image-input.styles.ts [hunk 1]
```diff
@@ -13,21 +17,21 @@ export const Wrapper = styled.div`
 		/* Generated background-image for the dashed border from https://kovart.github.io/dashed-border-generator/  */
 		return `background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg'%3e%3crect width='100%25' height='100%25' fill='none' rx='4' ry='4' stroke='${color}' stroke-width='4' stroke-dasharray='8%2c 8' stroke-dashoffset='8' stroke-linecap='square'/%3e%3c/svg%3e");`;
 	}}
```
Comment (weili-govtech) [inline src/components/fields/image-upload/image-input/image-input.styles.ts:19]: why remove the bottom margin for the wrapper? it is meant to space out this field from the next field.

File: src/components/fields/image-upload/image-input/image-input.tsx [hunk 2]
```diff
@@ -6,12 +6,20 @@ import { ImageContext } from "../image-context";
...
+	Subtitle,
+	UploadWrapper,
+	Wrapper,
+} from "./image-input.styles";
 
 interface IImageInputProps extends ISharedImageProps {
 	label: string;
 	buttonLabel?: string | undefined;
-	description: string;
+	description?: string;
```
Comment (weili-govtech) [inline src/components/fields/image-upload/image-input/image-input.tsx:22]:
```suggestion
	description?: string | undefined;
```

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/components/fields/image-upload/image-input/image-input.styles.ts (hunk 1 above)
Comment (Huizz) [reply src/components/fields/image-upload/image-input/image-input.styles.ts:19]: wrapper only has one child ([file](https://github.com/LifeSG/web-frontend-engine/blob/main/src/components/fields/image-upload/image-input/image-input.tsx#L162)) so I thought it was redundant

Comment (Huizz) [reply src/components/fields/image-upload/image-input/image-input.styles.ts:19]: oh this is not for the last child of Wrapper?

**Review by weili-govtech — APPROVED** [review]

**Pull request merged**